### ISSUE TITLE
FP: S1854 when assigned in try block and used in catch block

### ISSUE THREAD

**Issue by jasper-d** [SonarSource/sonar-dotnet#4071]
### Description

False positive S1854  when assigned in try and used in catch block

### Repro steps

```
  IDisposable subscription = null;
  Guid subscriptionId = Guid.Empty;  // FP S1854, removing the assignment will cause CS0165 : Use of unassigned local variable 'subscriptionId' when passing it to TryRemove
  try
  {
      subscription = doStuff();
      subscriptionId = AddSubscription(subscription);
  }
  catch (Exception)
  {
      _subscriptions.TryRemove(subscriptionId, out _);
      subscription?.Dispose();
      throw;
  }
```

### Expected behavior

Applying Sonar sugesstions shouldn't lead to invalid code.

### Actual behavior

Following the Sonar sugesstion results in invalid C#

### Known workarounds

Ignore S1854

### Related information

* MSBuild / dotnet version: 3.1
* C# Analyzer: 8.9 (build 19135)
* SonarScanner for .NET version (if used): SonarScanner for MSBuild 4.9
* Operating System: Linux

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jasper-d, 

Thank you for reporting this case. This is similar to #2766 and has the same root cause as #2600 so I'll close this issue. We plan to replace the CFG this year to fix all of them.

**Comment by megaromb** [comment]
> Hi @jasper-d,
> 
> Thank you for reporting this case. This is similar to #2766 and has the same root cause as #2600 so I'll close this issue. We plan to replace the CFG this year to fix all of them.

Hi @pavel-mikula-sonarsource, I confirm that as of today the issue hasn't been fixed in the extension version 8.5.0.10497.
The referenced quoted issues have been marked as "Closed" also.
So, you may have lost track of this issue.
My message is just to remind your team that the issue is still reproducible and needs to be fixed.
Thank you.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @megaromb,

Please create a new issue in this repository. Thank you
